oauth2_provider/oauth2_validators.py

```diff
@@ -753,18 +704,15 @@ def validate_refresh_token(self, refresh_token, client, request, *args, **kwargs
         return rt.application == client
 
     @transaction.atomic
-    def _save_id_token(self, token, request, expires, *args, **kwargs):
-
+    def _save_id_token(self, id_token, request, expires, *args, **kwargs):
         scopes = request.scope or " ".join(request.scopes)
 
-        if request.grant_type == "client_credentials":
-            request.user = None
-
         id_token = IDToken.objects.create(
+            token=id_token,
```

Comment:
**issue**: I think I needed to do `id_token.serializer()` here.
```suggestion
            token=id_token.serialize(),
```

Reply:
That makes sense I think as it is similar to what we have in the master branch.

Reply:
This was actually totally wrong, and need to be using the jwt token, not the id token. Fixed now.